src/exchanges/__tests__/crdtIntegration-test.js

```diff
@@ -0,0 +1,567 @@
+        }
+    },
+    _timestampedVariables: {
+        inspectionInput: {
+            inspection: timestampedInspectionCreateBase
+        },
+    }
+};
+
+const inspectionUpdateBase = {
+    uuid: inspectionUUID,
+    name: "updatedName",
+}
+const inspectionUpdateQueryResult = cloneDeep(inspectionCreateQueryResult);
+inspectionUpdateQueryResult.name = "updatedName";
+inspectionUpdateQueryResult.timestamps.name = expect.any(String);
+
+const inspectionUpdateMutationResult = cloneDeep(inspectionUpdateQueryResult);
+inspectionUpdateMutationResult.__typename = "Inspection";
+inspectionUpdateMutationResult.timestamps.__typename = "InspectionsTimestamp";
+
+const coalescedVariables = cloneDeep(timestampedInspectionCreateBase);
+coalescedVariables.name = "updatedName";
+coalescedVariables.timestamps.name = expect.any(String);
+
+const updateInspection = {
+    query: CreateOrUpdateInspection,
+    variables: {
+        inspectionInput: {
+            inspection: inspectionUpdateBase,
+        }
+    },
+    serverUpdateResult: {
+        createOrUpdateInspection: {
+            inspection: inspectionUpdateMutationResult,
+        }
+    },
+    _coalescedVariables: {
+        inspectionInput: {
+            inspection: coalescedVariables
+        }
+    }
+};
+
+const AllInspections = {
+    query: getAllInspectionsQuery,
+    _emptyArrayResult: {
+        allInspections: []
+    },
+    _createResult: {
+        allInspections: [inspectionCreateQueryResult]
+    },
+    _updateResult: {
+        allInspections: [inspectionUpdateQueryResult]
+    }
+}
+
+const nonCrdtMutationQuery = gql` 
+  mutation UpdateAuthor ($author: AuthorInput!) {
+    updateAuthor (author: $author) {
+      id
+      name
+    }
+  }
+`;
+
+const nonCrdtMutationData = {
+    __typename: 'Mutation',
+    updateAuthor: {
+        __typename: 'Author',
+        id: '123',
+        name: 'Me',
+    },
+};
+
+
+describe('crdtExchange', () => {
+    let client, ops$, next;
+    let triggerSend, writeMetadataCalled, resolveWriteMetadataCalled;
+    let localHlc;
+    let nonCrdtMutation, createInspectionMutation, updateInspectionMutation;
+    let getAllInspectionsQuery;
+    let timestampExchangeInstance, crdtExchangeInstance, offlineExchangeInstance, composedExchange;
+    const isRetryableError = jest.fn(() => true); // TODO import this from whereever this gets defined
+    const [getAllInspectionsQueryKey, nonCrdtMutationKey, createInspectionMutationKey, updateInspectionMutationKey] = [0,1,2,3]
+    const storage = {
+        onOnline: jest.fn(),
+        writeData: jest.fn(),
+        writeMetadata: jest.fn(),
+        readData: jest.fn(),
+        readMetadata: jest.fn(),
+    };
+
+    beforeEach(() => {
+        resetJestTime();
+        localHlc = new HLC(node, now);
+
+        [writeMetadataCalled, resolveWriteMetadataCalled] = unwrappedPromise();
+        storage.writeMetadata.mockImplementation(resolveWriteMetadataCalled);
```

Comment:
For an integration test, it seems a little bit too much of 'white box' testing to use this call to synchronize the tests. This could make this test suite more brittle than it really needs to be.

However, I'm ok with incurring this small bit of technical debt because it makes the tests very easy to read, and we have tons of higher value things to work on to make this project succeed.